kedro-datasets/kedro_datasets/_utils/file_utils.py

```diff
@@ -0,0 +1,18 @@
+import os
+
+
+def parse_glob_pattern(pattern: str) -> str:
+    special = ("*", "?", "[")
+    clean = []
+    for part in pattern.split("/"):
+        if any(char in part for char in special):
+            break
+        clean.append(part)
+    return "/".join(clean)
+
+
+def split_filepath(filepath: str | os.PathLike) -> tuple[str, str]:
+    split_ = str(filepath).split("://", 1)
+    if len(split_) == 2:  # noqa: PLR2004
+        return split_[0] + "://", split_[1]
+    return "", split_[0]
```

Comment:
AFAIK, these are used for Spark only. I do not want other datasets to use it because it's more like a workaround to handle Spark/Dtabricks magic path.

Most of the dataset should be using the util coming from `kedro` instead. https://github.com/kedro-org/kedro-plugins/blob/1a18c5a450a84cb582ea70739b108eb2edd54c29/kedro-datasets/kedro_datasets/pandas/csv_dataset.py#L17-L19

Reply:
@noklam Understood. I will move these to `spark_utils.py` module.